### DISCUSSION TITLE
X, Y, Z dimensions for shifts in motion correction

### DISCUSSION THREAD

**Discussion by ethanbb** [flatironinstitute/CaImAn#1430] [General]
I just realized, using a simple test case and then verifying by looking through the code a bit, that the shifts returned from motion correction are in (Y, X, Z) order, i.e. the indexing order assuming a properly constructed movie object or mmap file is passed in, rather than (X, Y, Z) order. In places where variables have x, y, or z in the name, e.g. `{x,y,z}_shifts_els`, it seems that X actually refers to Y and vice versa.

Is this intentional? It's not causing any bugs because it's used consistently that way within the file, but if someone wants to do something else with the shifts they could get confused. I know at least one mesmerize docstring needs to change based on this.

**Comment by kushalkolar** [comment]
they probably mixed up how cartesian coordinates are usually written in xy order but matrices are rows, cols which becomes y, x if you put an image on a cartesian plane

**Reply by ethanbb** [reply]
Right, of course...but we have to be careful to keep track of this

**Comment by ethanbb** [comment]
I realized that `tile_and_correct`  and `tile_and_correct_3d` contain some code with X and Y labeled correctly, mixing conventions (seemingly in a correct way, though it would be easy to get confused):

https://github.com/flatironinstitute/CaImAn/blob/bb55800806f0898592d79dcc705a0b53ccd01ec3/caiman/motion_correction.py#L2194-L2207

Here, `shift_img_x` and `shift_img_y` are swapped, but `x_grid` and `y_grid` actually refer to X and Y respectively, so you have `shift_img_y` added to `x_grid` and vice versa.
